### PR TITLE
feat(WAF): add datasource to query the list of the address group

### PR DESCRIPTION
**Pull request by MrLiangSir**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccDatasourceWAFAddressGroups_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDatasourceWAFAddressGroups_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceWAFAddressGroups_basic
=== PAUSE TestAccDatasourceWAFAddressGroups_basic
=== CONT  TestAccDatasourceWAFAddressGroups_basic
--- PASS: TestAccDatasourceWAFAddressGroups_basic (402.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       402.108s

### CODE REVIEW

File: docs/data-sources/waf_address_groups.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `groups` - A list of WAF address groups.
+
+The `groups` block supports:
+
+* `id` - The resource ID.
```
Comment (deer-hang) [inline docs/data-sources/waf_address_groups.md]:
```suggestion
* `id` - The WAF address group ID.
```

**Comment by MrLiangSir** [comment]
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccDatasourceWAFAddressGroups_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDatasourceWAFAddressGroups_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceWAFAddressGroups_basic
=== PAUSE TestAccDatasourceWAFAddressGroups_basic
=== CONT  TestAccDatasourceWAFAddressGroups_basic
    acceptance.go:441: Skip the WAF acceptance tests.
--- SKIP: TestAccDatasourceWAFAddressGroups_basic (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       0.040s

**Comment by MrLiangSir** [comment]
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccDatasourceWAFAddressGroups_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDatasourceWAFAddressGroups_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceWAFAddressGroups_basic
=== PAUSE TestAccDatasourceWAFAddressGroups_basic
=== CONT  TestAccDatasourceWAFAddressGroups_basic
--- PASS: TestAccDatasourceWAFAddressGroups_basic (413.68s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       413.727s

**Pull request merged**